Source/MixerButton.cpp

```diff
@@ -0,0 +1,139 @@
+//                         juce::Justification::centred, 1);
+//        g.drawFittedText("+20 dB", juce::Rectangle<int>(bounds.getRight() - 30, bounds.getY() + (getHeight() / 6) - 4,
+//                                                        20, 10),
+//                         juce::Justification::centred, 1);
+//    }
+
+    g.setFont (24.0f);
+    g.drawLine(70, getHeight() / 3, getWidth() - 20, getHeight() / 3, 1);
+    g.drawFittedText("dirty", bounds.removeFromTop(getHeight() / 3),
+                     juce::Justification::centred, 1);
+    g.drawLine(70, getHeight() / 3 * 2, getWidth() - 20, getHeight() / 3 * 2, 1);
+    g.drawFittedText("clean", bounds,
+                     juce::Justification::centred, 1);
+
+//    auto bubbleBounds = juce::Rectangle<float> (pointerPos_.getX() - 10, pointerPos_.getY() - 10, 20, 20);
+
+    // Selector Shadow
+//    g.setColour(juce::Colours::black.withAlpha(0.1f));
+//    g.fillEllipse(bubbleBounds.withPosition(bubbleBounds.getX(), bubbleBounds.getY() + 10));
+
+    // Selector
+//    g.setGradientFill(juce::ColourGradient(juce::Colours::grey.withAlpha(0.3f),
+//                         bubbleBounds.getCentreX(),
+//                         bubbleBounds.getCentreY(),
+//                         juce::Colours::white,
+//                         bubbleBounds.getX(), bubbleBounds.getY(), true)
+//                         );
+//    g.fillEllipse (bubbleBounds);
+    auto barBounds = juce::Rectangle<float> (sliderWidth, kHandleHeight)
+            .withCentre(bounds.getCentre().withY(pointerPos_.getY()).toFloat());
+    g.setGradientFill(juce::ColourGradient(juce::Colours::grey.withAlpha(0.3f),
+                         barBounds.getCentreX(),
+                         barBounds.getCentreY(),
+                         juce::Colours::white.withAlpha(0.7f),
+                         barBounds.getX() - (sliderWidth * 0.3f), barBounds.getY() + 2, true));
+    g.fillRoundedRectangle(barBounds, cornerRadius);
+    g.setColour (juce::Colours::white.withAlpha(0.3f));
+    g.drawRoundedRectangle(barBounds, cornerRadius, 1.0f);
+}
+
+void MixerButton::resized()
+{
+}
+
+void MixerButton::mouseDown(const juce::MouseEvent& event) {
+    auto bounds = getLocalBounds();
+
+    if (bounds.contains (event.getMouseDownPosition())) {
+        showdB = true;
+        mapMouseToValue(event.getMouseDownPosition());
+    }
+}
+
+void MixerButton::mouseDrag(const juce::MouseEvent& event) {
+    auto bounds = getLocalBounds();
+
+    if (bounds.contains (event.getMouseDownPosition())) {
+        showdB = true;
+        mapMouseToValue(event.getPosition());
+    }
+}
+
+void MixerButton::mouseUp(const juce::MouseEvent& event) {
+    auto bounds = getLocalBounds();
+
+    if (bounds.contains (event.getMouseDownPosition())) {
+        showdB = false;
+    }
+
+}
+
+void MixerButton::mapMouseToValue(const juce::Point<int>& mPoint)
+{
+    auto x_min = 70.0f;
+    auto x_max = getWidth() - 20.0f;
```

Comment:
using `auto` here makes it ambiguous if you want an `int` (getWidth()) or a `float`, easier to type the variable explicitly